### PR TITLE
Adding basic javadocs

### PR DESCRIPTION
**Pull request by mikaelm12**
Hoping to get some basic docs in for preview1.
Following the oracle outline here https://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#format

Fixes https://github.com/aspnet/SignalR/issues/2677

### CODE REVIEW

**Comment by mikaelm12** [comment]
Issue: https://github.com/aspnet/SignalR/issues/2677

File: clients/java/signalr/src/main/java/HubConnection.java [hunk 1]
```diff
@@ -90,22 +90,44 @@ public HubConnection(String url, Transport transport, Logger logger){
         }
     }
 
+    /**
+     * Initializes a new instance of the HubConnection class.
+     * @param url The url of the signalr server to connect to.
```
Comment (anurse) [inline clients/java/signalr/src/main/java/HubConnection.java]: nit: "SignalR" (capital 'S' and capital 'R')

File: clients/java/signalr/src/main/java/HubConnection.java [hunk 2]
```diff
@@ -90,22 +90,44 @@ public HubConnection(String url, Transport transport, Logger logger){
...
+     * @param transport The {@link Transport} that the client will use to communicate with the server.
+     */
     public HubConnection(String url, Transport transport) {
         this(url, transport, new NullLogger());
     }
 
+    /**
+     * Initializes a new instance of the HubConnection class.
+     * @param url The url of the signalr server to connect to.
```
Comment (anurse) [inline clients/java/signalr/src/main/java/HubConnection.java]: nit: "SignalR"

File: clients/java/signalr/src/main/java/HubConnection.java [hunk 3]
```diff
@@ -90,22 +90,44 @@ public HubConnection(String url, Transport transport, Logger logger){
...
+     */
     public HubConnection(String url) {
         this(url, null, new NullLogger());
     }
 
+    /**
+     * Initializes a new instance of the HubConnection class.
+     * @param url The url of the signalr server to connect to.
```
Comment (anurse) [inline clients/java/signalr/src/main/java/HubConnection.java]: nit: "SignalR"

File: clients/java/signalr/src/main/java/HubConnection.java [hunk 4]
```diff
@@ -90,22 +90,44 @@ public HubConnection(String url, Transport transport, Logger logger){
...
+     * Indicates the state of the {@link HubConnection} to the server.
+     * @return HubConnection state enum.
+     */
     public HubConnectionState getConnectionState() {
         return connectionState;
     }
 
+    /**
+     * Starts a connection to the server.
+     * @throws Exception If there was an error while connecting.
```
Comment (anurse) [inline clients/java/signalr/src/main/java/HubConnection.java]: Suggest: `@throws Exception An error occurred while connecting.`

File: clients/java/signalr/src/main/java/HubProtocol.java [hunk 5]
```diff
@@ -1,11 +1,26 @@
...
+
+    /**
+     * Creates a new list of {@link HubMessage}s
+     * @param message A string representation of one or more {@link HubMessage}s
+     * @return
+     */
     HubMessage[] parseMessages(String message);
+
+    /**
+     * Writes the specified {@link HubMessage}to a String.
```
Comment (anurse) [inline clients/java/signalr/src/main/java/HubProtocol.java]: nit: space after "HubMessage"

File: clients/java/signalr/src/main/java/HubConnection.java [hunk 6]
```diff
@@ -90,22 +90,44 @@ public HubConnection(String url, Transport transport, Logger logger){
...
+    /**
+     * Initializes a new instance of the HubConnection class.
+     * @param url The url of the signalr server to connect to.
+     */
     public HubConnection(String url) {
         this(url, null, new NullLogger());
     }
 
+    /**
+     * Initializes a new instance of the HubConnection class.
```
Comment (BrennanConroy) [inline clients/java/signalr/src/main/java/HubConnection.java]: Link the `HubConnection` in all these logs?

File: clients/java/signalr/src/main/java/HubConnection.java [hunk 7]
```diff
@@ -217,6 +375,10 @@ public Subscription on(String target, Action callback) {
         return new Subscription(handlers, action, target);
     }
 
+    /**
+     * Removes all handlers associated with the method with the specified method name.
+     * @param name The name of the hub method from which handlers are being removed
```
Comment (BrennanConroy) [inline clients/java/signalr/src/main/java/HubConnection.java]: nit: full stop

File: clients/java/signalr/src/main/java/HubProtocol.java [hunk 8]
```diff
@@ -1,11 +1,26 @@
...
+ * A protocol abstraction for communicating with SignalR hubs.
+ */
 public interface HubProtocol {
     String getName();
     int getVersion();
     TransferFormat getTransferFormat();
+
+    /**
+     * Creates a new list of {@link HubMessage}s
+     * @param message A string representation of one or more {@link HubMessage}s
```
Comment (BrennanConroy) [inline clients/java/signalr/src/main/java/HubProtocol.java]: nit: full stop

File: clients/java/signalr/src/main/java/HubProtocol.java [hunk 9]
```diff
@@ -1,11 +1,26 @@
...
+     * @return
```
Comment (BrennanConroy) [inline clients/java/signalr/src/main/java/HubProtocol.java]: Missed return text?

**Pull request merged**